libraries/AP_FlashIface/AP_FlashIface_JEDEC.cpp

```diff
@@ -162,6 +162,10 @@ void AP_FlashIface_JEDEC::reset_device()
 {
     // Get chip out of XIP mode
     AP_HAL::QSPIDevice::CommandHeader cmd;
+#ifndef HAL_BOOTLOADER_BUILD
+    _dev->get_semaphore()->take_blocking();
```

Comment:
need a comment here. why never released? why not in bootloader?

Reply:
Needed for the the test, comment added.